### ISSUE TITLE
useActor with @xstate/inspect crashing react app

### ISSUE THREAD

**Issue by dhirjgupta** [davidkpiano/xstate#1735]
## Description: 
TLDR: When passing a react ref into an actor via event, it breaks the inspector and causes a fatal error in the main react app as well. If I turn the inspector off, the react app works without crashing.

There is a machine, `Root`. It spawns several actors that will control UI elements, and stores them in context. This machine is connected to a react container component via `useMachine`.

```
import {rootMachine} from './rootMachine'

export default ContainerComponent = (props) => {
  const [state, send] = useMachine(rootMachine, {
    devTools: true
  });
  React.useEffect(() => {
     send('INIT'); // causes machine to initialize, spawning actors and storing in context
  }, []);
  return {
    <div>
      {state.context.actors['ui-element-1-id'] && 
        <UIElement1 actor={state.context.actors['ui-element-1-id']} />
      }
    </div>
  }
}
```

UIElement1 has several animations, triggered via services. The react ref is injected via an event, and referenced in the services by context.

```
export const UIElement1 = ({actor}) => {
  const myRef = React.useRef();
  const [state, send] = useActor(actor); // same result with useService, but with TS errors
  // if using machine and useMachine it works, but lose 'sendParent' functionality from actor

  React.useEffect(() => {
    send({type: 'INJECT_REACT_REF', ref: myRef})'
  }), []);

  return <div className="ui-element-1" ref={myRef} />
}
```

If I use `useMachine`, instead of `useActor` or `useService`, there is no error in the inspector. However, this causes the loss of the `sendParent` functionality.

The error received when using devtools in this situation is:

```
Uncaught TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'SVGPathElement'
    |     property '__reactInternalInstance$ibota64igp' -> object with constructor 'FiberNode'
    --- property 'stateNode' closes the circle
    at JSON.stringify (<anonymous>)
    at Interpreter.inspectSend [as send] (index.js:162) //  references @xstate/inspect/es/index.js
    at useActor.js:33
    at mech.tsx:35 // references the sending of the `INJECT_REACT_REF` line
```

If I turn the devtools off, the react application itself seems to work fine

## (Bug) Expected result:
inspector and react app should work

## (Bug) Actual result:
inspector and react app does not work

## (Bug) Potential fix:
No ideas unfortunately... However, its worth pointing out that using useMachine instead of useActor does not cause the breakage

## Link to reproduction or proof-of-concept:
https://codesandbox.io/s/serene-fermat-5nnd2?file=/src/mechs/Root.jsx
note - the sandbox example react app does not appear to break down in the same way as it does for me locally, and it does not throw error messages. But it does appear to break down fairly consistently.

The app is simple. Click the green square and it should animate to a selected look. Click again and it reverts. If you comment out the devtools in Root.jsx, it works. If you add it back in, things start breaking.

**Comment by Andarist** [comment]
I believe this has been fixed in `@xstate/inspect@0.3.0`. If not please ping me and I will reopen.

**Comment by dhirjgupta** [comment]
I found a functional workaround. When injecting the reactRef into the actor, dont inject directly. Instead inject a getter function.
```
export const myComponent = (props) => {
  const myRef = useRef();
  useEffect(() => {
    send({type: "INJECT", payload: { prop: "getRef", value: () => myRef}):
  }, [])
}
```

**Comment by kantorv** [comment]
> I found a functional workaround. When injecting the reactRef into the actor, dont inject directly. Instead inject a getter function.
> 
> ```
> export const myComponent = (props) => {
>   const myRef = useRef();
>   useEffect(() => {
>     send({type: "INJECT", payload: { prop: "getRef", value: () => myRef}):
>   }, [])
> }
> ```
> 
> Then access the getter via context in the animation services
> 
> ```
> const services = {
>   enterAnimation: (context, event) => {
>      const ref = context.getRef().current;
>     // animation code
>   }
> }
> ```

@dhirjgupta  can you please provide some codepen with workaround? playing a few days on issue, still have no idea how to use inspector and pass ref objects to the machine, every time getting the `Uncaught TypeError: Converting circular structure to JSON ...` 
have some workaround with passing `id` of element, then getting it with `getElementById`, but if there is a more suitable solution, whould like to try it out.. 

Thank You.

**Comment by kantorv** [comment]
found some workaround here, works on html elements, not on refs, but seems like it's a minor issue
[https://github.com/statelyai/xstate/issues/2048#issuecomment-836944939](https://github.com/statelyai/xstate/issues/2048#issuecomment-836944939)
